File: lintcode/linked-list-cycle.cpp
/**
 *  * Definition of ListNode
 *   * class ListNode {
 *    * public:
 *     *     int val;
 *      *     ListNode *next;
 *       *     ListNode(int val) {
 *        *         this->val = val;
 *         *         this->next = NULL;
 *          *     }
 *           * }
 *            */
class Solution {
		public:
		    /**
			 *      * @param head: The first node of linked list.
			 *           * @return: True if it has a cycle, or false
			 *                */
			    bool hasCycle(ListNode *head) {
					if(head == NULL || head->next == NULL) return false;
					ListNode *slow = head;
					ListNode *fast = head->next;
					while(fast){
						if(slow == fast) return true;
						slow = slow->next;
						if(!fast->next) return false;
						fast = fast->next->next;
					}
					return false;
		       }
};

File: pypractise/leetcode/linked-list-cycle.py
__author__ = 'Daniel.Lee'
class Solution:
    # @param head, a ListNode
    # @return a boolean
    def hasCycle(self, head):
        flag = head
        if head:
            if not head.next:
                return False
            head = head.next
        else:
            return False
        while head :
            if head == flag:
                return True
            head = head.next
        return False